### PR TITLE
Create price history entry

### PR DESCRIPTION
**Pull request by Kobe16**
I made the create price history entry endpoint for our API. The only weird part was trying to test the Date field because the response would return a string version of the Date. But then, if you used .toString() on our original Date object, it would return the Date, but in a different format. So, I had to work a little around that (i.e., the code right before the expect statements)

### CODE REVIEW

File: api/src/dtos/priceHistory.dto.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
-import { IsEmail, IsString, IsNotEmpty, MinLength, MaxLength, IsNumber, IsArray, IsDate } from 'class-validator';
+import { IsString, IsNotEmpty, MinLength, IsNumber, IsDate } from 'class-validator';
+import { ObjectId } from 'mongoose';
 
 export class CreatePriceHistoryDto {
 
-    @IsDate()
-    @IsNotEmpty()
-    public date: Date
+    // @IsString()
```
Comment (jbrw1984) [inline api/src/dtos/priceHistory.dto.ts]: I think you still need the `@IsString()` for the validation library.

File: api/src/models/priceHistory.model.ts [hunk 2]
```diff
@@ -10,6 +10,11 @@ const PriceHistorySchema: Schema = new Schema({
   //   required: true,
   //   unique: true,
   // },
+  pokemonCardId: {
```
Comment (jbrw1984) [inline api/src/models/priceHistory.model.ts:13]: Like this approach

File: api/src/routes/cards.route.ts [hunk 3]
```diff
@@ -15,11 +16,16 @@ export class CardsRoute implements Routes {
...
 
   private initializeRoutes() {
-    this.router.get(`${this.path}`, this.card.getCards);
-
     // Route for post request for creating a new card
     this.router.post(`${this.path}`, ValidationMiddleware(CreateCardDto), this.card.createCard);
 
+
+    // Route for post request for creating a new price history entry
+    this.router.post(`${this.path}/price-history`, ValidationMiddleware(CreatePriceHistoryDto), this.card.createPriceHistory)
```
Comment (jbrw1984) [inline api/src/routes/cards.route.ts]: Price history path should be a sub path of the pokemon card, like `$this.path/:id/price-history`, that way you can infer the pokenmon card ID and won't have to have it in the payload.

**Review by jbrw1984 — DISMISSED** [review]

**Review by averytribbett — APPROVED** [review]
Looks good. There might be some conflicts with the cards.test but other than that I see you made all the requested changes.

**Pull request merged**